docs/csharp/quick-starts/arrays-and-collections.md

```diff
@@ -0,0 +1,155 @@
+
+```csharp
+Console.WriteLine($"My name is {names[0]}");
+Console.WriteLine($"I've added {names[2]} and {names[3]} to the list");
+```
+
+It's not allowed to access past the end of the list. You can check how long the list is using the `<xref:System.Collections.Generic.List%601.Count%2A>` property. Add the following code at the end of the Main method:
+
+```csharp
+Console.WriteLine($"The list has {names.Count} people in it");
+ ```
+
+Save the file, and type `dotnet run` again to see the results.    
+
+## Search lists
+Our samples use relatively small lists, but in your applications you may often create lists with many more elements, sometimes numbering well over the thousands. To find elements in these larger collections, you need to search the list for different items. The <xref:System.Collections.Generic.List%601.IndexOf%2A> method searches for an item and returns the index of the item. Add this code to the bottom of your `Main` method:
+
+```csharp
+var index = names.IndexOf("Felipe");
+Console.WriteLine($"The name {names[index]} is at index {index}");
+
+var notFound = names.IndexOf("Not Found");
+Console.WriteLine($"When an item is not found, IndexOf returns {notFound}");
+```
+
+The items in your list can be sorted as well. The <xref:System.Collections.Generic.List%601.Sort%2A> method sorts all the items in the list in their normal order (alphabetically in the case of strings). Add this code to the bottom of our `Main` method:
+
+```csharp
+names.Sort();
+foreach (var name in names)
+{
+    Console.WriteLine($"Hello {name.ToUpper()}!");
+}
+```
+
+Save the file and type `dotnet run` to try this latest version.
+
+Before you start the next section, let's move the current code into a separate method. That makes it easier to start working with a new example. Rename your `Main` method to `WorkingWithStrings` and write a new `Main` methdo that calls `WorkingWithStrings`. When you have finished, your code should look like this:
+
+[!code-csharp[WorkingWithStrings](../../../samples/csharp/list-quickstart/program.cs#WorkingWithStrings "Working with Strings")]
+
+
+## Lists of other types
+
+You've been using the `string` type in lists so far. Let's make a <xref:System.Collections.Generic.List%601> using a different type. Let's build a set of numbers. 
+
+Add the following to the bottom of your new `Main` method:
+
+```csharp
+var fibonacciNumbers = new List<int> {1, 1,};
+```
+
+That creates a list of integers, and sets the first two integers to the value 1. These are the first two values of a *Fibonacci Sequence*, a sequence of numbers. Each next Fibonacci number is found by taking the sum of the previous two numbers. Add this code:
+
+```csharp
+var previous = fibonacciNumbers[fibonacciNumbers.Count - 1];
+var previous2 = fibonacciNumbers[fibonacciNumbers.Count - 2];
+
+fibonacciNumbers.Add(previous + previous2);
+
+foreach(var item in fibonacciNumbers)
+    Console.WriteLine(item);
+```
```

Comment:
The code needs a little bit more explanation -- with each iteration of the loop, you're taking the last two integers in the list, summing them, and adding that value to the list, and then repeating the procedure.